src/main/java/com/shapesecurity/functional/data/ImmutableSet.java

```diff
@@ -74,6 +90,15 @@ public boolean contains(@Nonnull T datum) {
         return new ImmutableSet<>(this.data.merge(other.data));
     }
 
+    @Nonnull
+    public ImmutableSet<T> union(@Nonnull Set<T> other) {
```

Comment:
This can take an `Iterable`.

There's an interesting design question here: for some of the other standard set operations (i.e., union, intersection, difference, symmetric difference) you can't take an iterable without big-O overhead. But in this case you can, and I think should even though that means the signatures of these methods will be inconsistent with each other if we ever add the others.